lib/api/core/plugins/pluginContext.js

```diff
@@ -75,23 +76,67 @@ class PluginContext {
 
     if (kuzzle instanceof Kuzzle) {
       const
-        pluginIndex = `plugin:${pluginName}`.toLowerCase(), // uppercase are forbidden by ES
-        // @todo use internalStorage
-        pluginInternalEngine = {
-          init: () => 'mock',
-          createCollection: () => 'mock',
-          bootstrap: {
-            startOrWait: () => 'mock'
-          }
-        },
-        kuzzleSdk = new KuzzleSDK(new FunnelProtocol(kuzzle.funnel));
-
-      pluginInternalEngine.init(
-        new PluginBootstrap(
-          pluginName,
+        kuzzleSdk = new KuzzleSDK(new FunnelProtocol(kuzzle.funnel)),
+        // uppercase are forbidden by ES
+        pluginIndex = `plugin:${pluginName}`.toLowerCase(),
+        pluginIndexEngine = new IndexEngine(
           kuzzle,
-          pluginInternalEngine,
-          pluginIndex));
+          pluginIndex,
+          kuzzle.services.internalStorage);
+
+      // Set the bootstraper class to initialize plugin index
+      pluginIndexEngine.indexBootstap = new PluginIndexBootstrap(
+        kuzzle,
+        pluginName,
+        pluginIndexEngine);
+
+      Object.defineProperty(this.accessors, 'storage', {
+        enumerable: true,
+        get: () => {
+          return {
+            bootstrap: () => pluginIndexEngine.init(),
+            createCollection: (...args) => pluginIndexEngine.createCollection(...args)
+          };
+        }
+      });
+
+      /**
+       * @todo Use a CollectionEngine class instead repository
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)